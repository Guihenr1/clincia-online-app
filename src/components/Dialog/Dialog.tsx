import { FC } from "react";
import useStyles from "./styles";

import DialogMui from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogTitle from "@mui/material/DialogTitle";
import Button from "../Button/Button";

interface DialogProps {
  open: boolean;
  title: string;
  content: string;
  handleClose: () => void;
  handleConfirm: () => void;
}

const Dialog: FC<DialogProps> = ({
  open,
  title,
  content,
  handleClose,
  handleConfirm,
}) => {
  const s = useStyles();

  return (
    <DialogMui open={open} onClose={handleClose}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText>{content}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose}>Disagree</Button>
        <Button onClick={handleConfirm}>Agree</Button>
      </DialogActions>
    </DialogMui>
  );
};

Dialog.defaultProps = {
  title: "",
  content: "",
};

export default Dialog;
