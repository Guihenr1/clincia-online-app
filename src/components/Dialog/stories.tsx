import { Meta } from "@storybook/react";
import Dialog from "./Dialog";

export default {
  component: Dialog,
  args: {
    title: "Do you confirm?",
    content: `Lorem ipsum dolor sit amet, 
      consectetur adipiscing elit. Etiam eget ligula eu lectus lobortis condimentum. 
      Aliquam nonummy auctor massa. Pellentesque habitant morbi tristique senectus et 
      netus et malesuada fames ac turpis egestas. Nulla at risus. Quisque purus magna, 
      auctor et, sagittis ac, posuere eu, lectus. Nam mattis, felis ut adipiscing`,
  },
} as Meta;

export const Primary = (args: any) => {
  return <Dialog {...args} />;
};

Primary.args = {
  open: true,
};
