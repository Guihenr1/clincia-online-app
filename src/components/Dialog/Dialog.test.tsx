import { fireEvent, render, screen } from "@testing-library/react";
import Dialog from "./Dialog";
import "@testing-library/jest-dom/extend-expect";

describe("Dialog", () => {
  const title = "Title of this";
  const content = "Text";
  const handleConfirm = () => {};
  const handleClose = () => {};

  test("renders the Dialog", () => {
    const { container } = render(
      <Dialog
        open={true}
        content={content}
        title={title}
        handleConfirm={handleConfirm}
        handleClose={handleClose}
      />
    );

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(content)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
