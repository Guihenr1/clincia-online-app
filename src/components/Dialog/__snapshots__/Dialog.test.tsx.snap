// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders the Dialog 1`] = `
<div
  aria-hidden="true"
/>
`;
